bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DateTimeType.java

```diff
@@ -89,18 +92,23 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
-        if (this == obj)
+        if (this == obj) {
             return true;
-        if (obj == null)
+        }
+        if (obj == null) {
             return false;
-        if (getClass() != obj.getClass())
+        }
+        if (getClass() != obj.getClass()) {
             return false;
+        }
         DateTimeType other = (DateTimeType) obj;
         if (calendar == null) {
-            if (other.calendar != null)
+            if (other.calendar != null) {
                 return false;
-        } else if (!calendar.equals(other.calendar))
+            }
+        } else if (calendar.compareTo(other.calendar) != 0) {

```

Comment:
> Do also mean that we should move in the direction to make DateTimeType immutable?

I don't have an opinion on that if `compareTo` is used, but it might make sense to make `DateTimeType` consistent with all other types.  @kaikreuzer, your opinion?